components/user_settings/display/manage_languages/manage_languages.tsx

```diff
@@ -1,144 +1,176 @@
-                } else if (err) {
-                    let serverError;
-                    if (err.message) {
-                        serverError = err.message;
-                    } else {
-                        serverError = err;
-                    }
-                    this.setState({serverError, isSaving: false});
+        this.props.actions.updateMe(user).then((res) => {
+            if ('data' in res) {
+                // Do nothing since changing the locale essentially refreshes the page
+            } else if ('error' in res) {
+                let serverError;
+                const {error} = res;
+                if (error instanceof Error) {
+                    serverError = error.message;
+                } else {
+                    serverError = error;
                 }
-            });
-    }
+                this.setState({serverError, isSaving: false});
+            }
+        });
+    };
 
     handleMenuClose = () => {
-        document.querySelector('.modal-body').classList.remove('no-scroll');
+        const modalBody = document.querySelector('.modal-body');
+        if (modalBody) {
+            modalBody.classList.remove('no-scroll');
+        }
         this.setState({openMenu: false});
-    }
+    };
 
     handleMenuOpen = () => {
-        document.querySelector('.modal-body').classList.add('no-scroll');
+        const modalBody = document.querySelector('.modal-body');
+        if (modalBody) {
+            modalBody.classList.add('no-scroll');
+        }
         this.setState({openMenu: true});
-    }
+    };
 
     render() {
         let serverError;
         if (this.state.serverError) {
-            serverError = <label className='has-error'>{this.state.serverError}</label>;
+            serverError = (
+                <label className='has-error'>{this.state.serverError}</label>
+            );
         }
 
-        const options = [];
-        const locales = I18n.getLanguages();
+        const options: object[] = [];
+        const locales: any = I18n.getLanguages();
 
-        const languages = Object.keys(locales).map((l) => {
-            return {
-                value: locales[l].value,
-                name: locales[l].name,
-                order: locales[l].order,
-            };
-        }).sort((a, b) => a.order - b.order);
+        const languages = Object.keys(locales).
+            map((l) => {
+                return {
+                    value: locales[l].value as string,
+                    name: locales[l].name,
+                    order: locales[l].order
+                };
+            }).
+            sort((a, b) => a.order - b.order);
 
         languages.forEach((lang) => {
-            options.push(
-                {value: lang.value, label: lang.name},
-            );
+            options.push({value: lang.value, label: lang.name});
         });
 
         const reactStyles = {
-            menuPortal: (provided) => ({
+            menuPortal: (provided: any) => ({
```

Comment:
I believe there is a type for styles in React. Should be something like `CSS.Properties`

Reply:
resolved